grpc/src/tx.rs

```diff
@@ -244,15 +238,12 @@ where
     }
 
     /// Get most recent minimal gas price seen by the client
-    pub fn last_seen_gas_price(&self) -> f64 {
-        *self.gas_price.read().expect("lock poisoned")
+    pub async fn get_min_gas_price(&self) -> Result<f64> {
+        self.client.get_min_gas_price().await
     }
 
-    /// Set current gas price used by the client
-    async fn update_gas_price(&self) -> Result<f64> {
-        let gas_price = self.client.get_min_gas_price().await?;
-        *self.gas_price.write().expect("lock poisoned") = gas_price;
-        Ok(gas_price)
+    pub async fn get_estimate_gas_price(&self, priority: TxPriority) -> Result<f64> {
+        self.client.get_estimate_gas_price(priority).await
```

Comment:
Since we're moving to a place where TxClient does not do any gas estimation, I've removed the min gas price caching from it. This means that `last_seen_gas_price` no longer makes sense, but I thought I'd add these two as a substitute. Alternative would be expecting user to create a `GrpcClient` and query the appropriate functions there directly.
wdyt? @zvolin @oblique